core/ecschema-metadata/src/Deserialization/Helper.ts

```diff
@@ -1010,11 +1010,18 @@ export class SchemaReadHelper<T = unknown> {
   private async loadCustomAttributes(container: AnyCAContainer, caProviders: Iterable<CAProviderTuple>): Promise<void> {
     for (const providerTuple of caProviders) {
       // First tuple entry is the CA class name.
-      const caClass = await this.findSchemaItem(providerTuple[0]) as CustomAttributeClass;
+      const caClass = await this.findSchemaItem(providerTuple[0]);
+
+      // If custom attribute exist within the context and is referenced, validate the reference is defined in the container's schema
+      if (caClass && caClass.key.schemaName !== container.schema.name &&
+        !container.schema.references.find((reference) => reference.name === caClass.key.schemaName)) {
```

Comment:
you should use Schema.getReferenceSync here instead of using find directly.  if you look at the implementation you'll notice that it is doing a lower case comparison while you are comparing by case

Reply:
Changed to use getReferenceSync.